api/src/pcapi/alembic/versions/20231208T130817_c0c6a91a03cb_make_ban_id_non_nullable_for_permanent_ids.py

```diff
@@ -0,0 +1,31 @@
+"""
+Check that Ban ID is not null for permanent venues
+"""
+from alembic import op
+
+
+# pre/post deployment: post
+# revision identifiers, used by Alembic.
+revision = "c0c6a91a03cb"
+down_revision = "041c43f6659c"
+branch_labels = None
+depends_on = None
+
+
+CONSTRAINT_CHECK_IS_PERMANENT_AND_HAS_BAN_ID_OR_IS_VIRTUAL = """
+    ("isPermanent" IS FALSE)
+    OR ("isVirtual" IS True) 
+    OR ("isPermanent" IS TRUE AND "isVirtual" IS FALSE AND "banId" IS NOT NULL)
+"""
+
+
+def upgrade() -> None:
+    op.create_check_constraint(
```

Comment:
L'ajout d'une contrainte sur une table existante requiert un scan de toute la table en prenant un SHARE ROW EXCLUSIVE qui empêche les écritures en DB. On préfère donc faire ce type de migration en 2 temps: 
- ajout de la contrainte NOT VALID (pas de scan pour vérifier chaque rang avec la contrainte)
- validation de la contrainte

Dans le cas présent, la table venue ne contient pas trop de lignes et n'a pas besoin de beaucoup d'écritures (par rapport à la table booking par exemple), mais il faudra surveiller le temps que cela prend lorsque la migration sera jouée en `staging`.

Reply:
On a justement un script qui génère les (4) bonnes migrations nécessaires à la mise en place d'une contrainte NOT NULL sur une table volumineuse et/ou fréquemment modifiée :

    python bin/alembic_add_not_null_constraint --table TABLE --column COLUMN

C'est facile à utiliser et ça fait des migrations sans douleur ni downtime. :)

Reply:
Je suis hors sujet, cette PR n'ajoute pas une contrainte NOT NULL. (Ceci dit, on pourrait réutiliser la sortie de ce script pour créer sereinement la contrainte souhaitée dans cette PR.)